src/engine/BMGame.php

```diff
@@ -939,8 +821,10 @@ protected function roll_active_dice_needing_values() {
 
     protected function update_game_state_specify_dice() {
         if (!$this->isWaitingOnAnyAction()) {
-            $this->prevSwingValueArrayArray = NULL;
-            $this->prevOptValueArrayArray = NULL;
+            foreach ($this->playerArray as $player) {
+                $player->prevSwingValueArray = array();
+                $player->prevOptValueArray = array();
+            }
```

Comment:
Before, we had to do *two* checks:
- is it an array?
- is the array empty?

The shift from NULL to array() gets around the first check.

I'm not going to excuse PHP for being terrible!